modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -127,6 +127,11 @@ public ObjectField deleteObjectField(long objectFieldId)
 		ObjectField objectField = objectFieldPersistence.findByPrimaryKey(
 			objectFieldId);
 
+		if (Validator.isNotNull(objectField.getRelationshipType())) {
+			throw new ObjectFieldRelationshipTypeException.
+				MustNotDeleteObjectFieldRelationshipType();
+		}
+
```

Comment:
@marcelabc this check should be inside the `deleteObjectField(ObjectField objectField)`. That way, we ensure that we don't overpass this validation when someone deletes any field.